llvm/lib/Target/AArch64/AArch64ISelLowering.cpp

```diff
@@ -2105,15 +2105,12 @@ bool AArch64TargetLowering::shouldExpandGetActiveLaneMask(EVT ResVT,
   if (!Subtarget->hasSVE())
     return true;
 
-  // We can only support legal predicate result types. We can use the SVE
-  // whilelo instruction for generating fixed-width predicates too.
-  if (ResVT != MVT::nxv2i1 && ResVT != MVT::nxv4i1 && ResVT != MVT::nxv8i1 &&
-      ResVT != MVT::nxv16i1 && ResVT != MVT::v2i1 && ResVT != MVT::v4i1 &&
-      ResVT != MVT::v8i1 && ResVT != MVT::v16i1)
+  if (!ResVT.isVector() || ResVT.getVectorElementType() != MVT::i1)
```

Comment:
Calling this function with a non-vector `ResVT` is just a bug, so you can drop the `!ResVT.isVector() ||` part and leave `getVectorElementType()` to raise an assert if that happens.